beacon-chain/cache/committee.go

```diff
@@ -52,14 +56,16 @@ func committeeKeyFn(obj interface{}) (string, error) {
 func NewCommitteesCache() *CommitteeCache {
 	return &CommitteeCache{
 		CommitteeCache: lruwrpr.New(int(maxCommitteesCacheSize)),
+		inProgress:     make(map[string]bool),
 	}
 }
 
 // Committee fetches the shuffled indices by slot and committee index. Every list of indices
 // represent one committee. Returns true if the list exists with slot and committee index. Otherwise returns false, nil.
-func (c *CommitteeCache) Committee(slot types.Slot, seed [32]byte, index types.CommitteeIndex) ([]types.ValidatorIndex, error) {
-	c.lock.RLock()
-	defer c.lock.RUnlock()
+func (c *CommitteeCache) Committee(ctx context.Context, slot types.Slot, seed [32]byte, index types.CommitteeIndex) ([]types.ValidatorIndex, error) {
+	mLock := async.NewMultilock(lockKey(seed))
```

Comment:
lockKey function in this package is too generic. Seems like it can be used by anything, let's make it specific to the committee cache